app/Console/Commands/Indexer/IncrementalIndexer.php

```diff
@@ -0,0 +1,226 @@
+                {
+                    continue;
+                }
+
+                $this->error("[SKIPPED] Failed to fetch $url - {$response['error']}");
+            }
+            catch (\Exception)
+            {
+                $this->warn("[SKIPPED] Failed to fetch $url");
+                $failedIds[] = $id;
+                Storage::put("indexer/incremental/$mediaType.failed", json_encode($failedIds));
+            }
+
+            $success[] = $id;
+            Storage::put("indexer/incremental/{$mediaType}_resume.save", $index);
+        }
+
+        Storage::delete("indexer/incremental/{$mediaType}_resume.save");
+
+        $this->info("--- Indexing of $mediaType is complete.");
+        $this->info(count($success) . ' entries indexed or updated.');
+        if (count($failedIds) > 0)
+        {
+            $this->info(count($failedIds) . ' entries failed to index or update. Re-run with --failed to requeue failed entries only.');
+        }
+
+        // finalize the latest state
+        Storage::move("indexer/incremental/$mediaType.json.tmp", "indexer/incremental/$mediaType.json");
+    }
+
+    public function handle(): int
+    {
+        // validate inputs
+        $validator = Validator::make(
+            [
+                'mediaType' => $this->argument('mediaType'),
+                'delay' => $this->option('delay'),
+                'resume' => $this->option('resume') ?? false,
+                'failed' => $this->option('failed') ?? false
+            ],
+            [
+                'mediaType' => 'required|in:anime,manga',
+                'delay' => 'integer|min:1',
+                'resume' => 'bool|prohibited_with:failed',
+                'failed' => 'bool|prohibited_with:resume'
+            ]
+        );
+
+        if ($validator->fails()) {
+            $this->error($validator->errors()->toJson());
+            return 1;
+        }
+
+        // we want to handle signals from the OS
+        $this->trap(SIGTERM, fn () => $this->cancelled = true);
+
+        $resume = $this->option('resume') ?? false;
+        $onlyFailed = $this->option('failed') ?? false;
+
+        /**
+         * @var $mediaTypes array
+         */
+        $mediaTypes = $this->argument("mediaType");
+
+        foreach ($mediaTypes as $mediaType)
+        {
+            $idsToFetch = [];
+
+            // if "--failed" option is specified just run the failed ones
+            if ($onlyFailed && Storage::exists("indexer/incremental/{$mediaType}_failed.json"))
+            {
+                $idsToFetch["sfw"] = $this->getFailedIdsToFetch($mediaType);
+            }
+            else
+            {
+                $idsToFetch = $this->getIdsToFetch($mediaType);
+            }
+
+            $idCount = count($idsToFetch);
+            if ($idCount == 0)
+            {
+                if ($this->cancelled)
+                {
+                    return 127;
+                }
+                continue;
+            }
```

Comment:
Could this be simplified to this?

```php
if ($idCount === 0 && $this->cancelled) {
    return 127;
}
```

Reply:
Not really. If it's `0` it can still continue to the next `mediaType` to fetch. Rather the check for `cancelled` should be outside of the `if` statement. Will fix it tonight.
Also It should be `===` not `==`.

Reply:
fixed it